x/src/test.rs

```diff
@@ -19,13 +24,36 @@ pub struct Args {
     unit: bool,
     #[structopt(name = "TESTNAME", parse(from_os_str))]
     testname: Option<OsString>,
+    #[structopt(name = "LCOVFILE", parse(from_os_str))]
+    lcovfile: Option<OsString>,
+    #[structopt(name = "HTMLCOVDIR", parse(from_os_str))]
+    htmlcovdir: Option<OsString>,
     #[structopt(name = "ARGS", parse(from_os_str), last = true)]
     args: Vec<OsString>,
 }
 
 pub fn run(mut args: Args, config: Config) -> Result<()> {
     args.args.extend(args.testname.clone());
 
+    //if we will be be running coverage, set the environment arguments we'll force
+    let runcov = args.lcovfile.is_some() || args.htmlcovdir.is_some();
+    let env_vars: Vec<&str> = if runcov {
+        [
+            //when test fail try to tell the user why
+            "RUST_BACTRACE=1",
+            //A way to use -Z (unstable) flags with the stable compiler. See below.
+            "RUSTC_BOOTSTRAP=1",
+            //Recommend setting for grcov, avoids using the cargo cache.
+            "CARGO_ICREMENTAL=0",
+            //recommend flags for use with grcov, with Bob Willson's edit to remove: -Copt-level=0 already missing -Clink-dead-code
+            //for more info see:  https://github.com/mozilla/grcov#example-how-to-generate-gcda-fiels-for-a-rust-project
+            "RUSTFLAGS=\"-Zprofile -Ccodegen-units=1 -Coverflow-checks=off -Zno-landing-pads\"",
+        ]
+        .to_vec()
+    } else {
+        ["RUST_BACTRACE=1"].to_vec()
```

Comment:
I don't think its a good idea to have this turned on by default, everyone knows how to enable it if they need it